app/vmalert/notifier/faker.go

```diff
@@ -15,6 +17,18 @@ type FakeNotifier struct {
 	counter int
 }
 
+func InitFakeNotifier() *FakeNotifier {
+	fn := &FakeNotifier{}
+	getActiveNotifiers = func() []Notifier {
+		return []Notifier{fn}
+	}
+	return fn
+}
+
+func ResetFakeNotifier() {
+	getActiveNotifiers = nil
+}
```

Comment:
This seems weird... It suppose to reset fake notifiers according to the name. But in fact it resets the global variable. And this method is public, which is even more weird.